packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -37,20 +50,111 @@ const COPY_ICON_CLASS = 'elyra-copy-icon';
 const INSERT_ICON_CLASS = 'elyra-add-icon';
 
 /**
- * CodeSnippetTable props.
+ * CodeSnippetDisplay props.
  */
-interface ICodeSnippetProps {
+interface ICodeSnippetDisplayProps {
   codeSnippets: ICodeSnippet[];
+  getCurrentWidget: () => Widget;
 }
 
 /**
  * A React Component for code-snippets display list.
  */
-class CodeSnippetTable extends React.Component<ICodeSnippetProps> {
+class CodeSnippetDisplay extends React.Component<ICodeSnippetDisplayProps> {
   // TODO: Use code mirror to display code
-  // TODO: implement copy to clipboard command
-  // TODO: implement insert code to file editor command (first check for code language matches file editor kernel language)
 
+  // Handle code snippet insert into an editor
+  private insertCodeSnippet(snippet: ICodeSnippet): void {
+    const widget: Widget = this.props.getCurrentWidget();
+    const snippetStr: string = snippet.code.join('\n');
+
+    if (
+      widget instanceof DocumentWidget &&
+      (widget as DocumentWidget).content instanceof FileEditor
+    ) {
+      const documentWidget = widget as DocumentWidget;
+      const fileEditor = (documentWidget.content as FileEditor).editor;
+      const markdownRegex = /md|mkdn?|mdown|markdown/g;
+
+      if (PathExt.extname(widget.context.path).match(markdownRegex) !== null) {
+        // Wrap snippet into a code block when inserting it into a markdown file
+        fileEditor.replaceSelection(
+          '```' + snippet.language + '\n' + snippetStr + '\n```'
+        );
+      } else if (widget instanceof PythonFileEditor) {
+        this.verifyLanguageAndInsert(
+          snippet.language,
+          snippet.displayName,
+          'python',
+          fileEditor,
+          snippetStr
+        );
+      } else {
+        fileEditor.replaceSelection(snippetStr);
+      }
+    } else if (widget instanceof NotebookPanel) {
+      const notebookWidget = widget as NotebookPanel;
+      const notebookCellEditor = (notebookWidget.content as Notebook).activeCell
+        .editor;
+      const kernelLanguage: string =
+        notebookWidget.context.sessionContext.kernelPreference.language;
+      this.verifyLanguageAndInsert(
+        snippet.language,
+        snippet.displayName,
+        kernelLanguage,
+        notebookCellEditor,
+        snippetStr
+      );
+    } else {
+      this.showErrDialog('Code snippet insert failed: Unsupported widget');
+    }
+  }
+
+  // Handle language compatibility between code snippet and editor
+  private verifyLanguageAndInsert = async (
+    snippetLanguage: string,
+    snippetName: string,
+    editorLanguage: string,
+    editor: CodeEditor.IEditor,
+    code: string
```

Comment:
I think it might make more sense to pass an `ICodeSnippet` param instead of passing 3 values from it